test/smoke/framework/kubernetes.go

```diff
@@ -0,0 +1,150 @@
+	}
+	return nil
+}
+
+// DeleteNamespace deletes a namespace
+func DeleteNamespace(namespace string) error {
+	ns := &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: namespace}}
+	GetLogger(namespace).Infof("Delete namespace %s", namespace)
+	err := kubernetes.ResourceC(kubeClient).Delete(ns)
+	if err != nil {
+		return fmt.Errorf("Cannot delete namespace %s: %v", namespace, err)
+	}
+	return nil
+}
+
+// IsNamespace checks wherher a namespace exists
+func IsNamespace(namespace string) (bool, error) {
+	ns, err := kubernetes.NamespaceC(kubeClient).Fetch(namespace)
+	if err != nil {
+		return false, fmt.Errorf("Cannot create namespace %s: %v", namespace, err)
+	}
+	return ns != nil, nil
+}
+
+// WaitForPods waits for pods with specific label to be available and running
+func WaitForPods(namespace, labelName, labelValue string, numberOfPods, timeoutInMin int) error {
+	return WaitFor(namespace, fmt.Sprintf("Pods with label name '%s' and value '%s' available and running", labelName, labelValue), time.Duration(timeoutInMin)*time.Minute, func() (bool, error) {
+		pods, err := GetPods(namespace, map[string]string{labelName: labelValue})
+		if err != nil || (len(pods.Items) != numberOfPods) {
+			return false, err
+		}
+
+		return CheckPodsAreRunning(pods), nil
+	})
+}
+
+// GetPods retrieves pods based on label name and value
+func GetPods(namespace string, labels map[string]string) (*corev1.PodList, error) {
+	pods := &corev1.PodList{}
+	if err := kubernetes.ResourceC(kubeClient).ListWithNamespaceAndLabel(namespace, pods, labels); err != nil {
+		return nil, err
+	}
+	return pods, nil
+}
+
+// CheckPodsAreRunning returns true if all pods are running
+func CheckPodsAreRunning(pods *corev1.PodList) bool {
+	for _, pod := range pods.Items {
+		if pod.Status.Phase != corev1.PodRunning {
+			return false
+		}
+	}
+	return true
+}
+
+// WaitForStatefulSetRunning waits for a stateful set to be running, with a specific number of pod
+func WaitForStatefulSetRunning(namespace, ssName string, podNb int, timeoutInMin int) error {
+	return WaitFor(namespace, fmt.Sprintf("StatefulSet %s running", ssName), time.Duration(timeoutInMin)*time.Minute, func() (bool, error) {
+		if dc, err := GetStatefulSet(namespace, ssName); err != nil {
+			return false, err
+		} else if dc == nil {
+			return false, nil
+		} else {
+			GetLogger(namespace).Debugf("StatefulSet %s has %d ready replicas\n", ssName, dc.Status.ReadyReplicas)
+			return dc.Status.ReadyReplicas == int32(podNb), nil
+		}
+	})
+}
+
+// GetStatefulSet retrieves a stateful set
+func GetStatefulSet(namespace, ssName string) (*appsv1.StatefulSet, error) {
+	dc := &appsv1.StatefulSet{}
+	if exists, err := kubernetes.ResourceC(kubeClient).FetchWithKey(types.NamespacedName{Name: ssName, Namespace: namespace}, dc); err != nil && !errors.IsNotFound(err) {
+		return nil, fmt.Errorf("Error while trying to look for DeploymentConfig %s: %v ", ssName, err)
+	} else if errors.IsNotFound(err) || !exists {
```

Comment:
The `notFound` error is handled inside this function, that's why the `exists` flag is for. I guess you  won't need this function at all since you are going to do the same checks when calling it: for any errors and if the resource is not `nil` before using it.